assets/blocks/quiz/question-block/question-block-attributes.js

```diff
@@ -34,29 +48,149 @@ export function createQuestionBlockAttributes( question ) {
 
 /**
  * Type-specific block attributes.
+ *
+ * @param {Object} apiAttributes The REST API response for a question.
  */
-const getTypeAttributes = {
-	'gap-fill': ( { before, after, gap } ) => ( {
-		answer: {
-			textBefore: before,
-			textAfter: after,
-			rightAnswers: gap,
-		},
-	} ),
-	'multiple-choice': ( { options, random_order: randomOrder } ) => ( {
-		answer: {
-			answers: options.map( ( { label: title, correct: isRight } ) => ( {
-				title,
-				isRight,
-			} ) ),
-		},
-		options: {
-			randomOrder,
-		},
-	} ),
-	boolean: ( { answer: rightAnswer } ) => ( {
-		answer: {
-			rightAnswer,
-		},
-	} ),
+const getTypeAttributes = ( apiAttributes ) => {
+	let blockAttributes;
+
+	switch ( apiAttributes.type ) {
+		case 'gap-fill':
+			blockAttributes = {
+				answer: {
+					textBefore: apiAttributes.before,
+					textAfter: apiAttributes.after,
+					rightAnswers: apiAttributes.gap,
+				},
+			};
+			break;
+		case 'multiple-choice':
+			blockAttributes = {
+				answer: {
+					answers: apiAttributes.options.map(
+						( { label: title, correct: isRight } ) => ( {
+							title,
+							isRight,
+						} )
+					),
+				},
+				options: {
+					randomOrder: apiAttributes.random_order,
+				},
+			};
+			break;
+		case 'boolean':
+			blockAttributes = {
+				answer: {
+					rightAnswer: apiAttributes.answer,
+				},
+			};
+			break;
+		case 'single-line':
+		case 'multi-line':
+		case 'file-upload':
+			blockAttributes = {};
+			break;
+		default:
+			blockAttributes = {
+				answer: normalizeAttributes( apiAttributes, camelCase ),
```

Comment:
Why are we putting these in `answer` by default?

Reply:
So currently, a custom component for a custom question type would require an `edit` component and optionally a selection of component settings (check `answer-blocks/index.js`). The way this works in `question-edit` is that we pass everything under `answer` in the edit component and everything under `options` in the settings components.

If with the defaults we want to achieve a solution that might work, we need to pass it under `answer` otherwise no attributes will be passed to the edit component.

Reply:
I have tried to document the above a bit in the `sense_quiz_question_types` filter.

Reply:
Ahh cool. Thanks for the explanation.